### PR TITLE
Add feedback api

### PR DESCRIPTION
**Pull request by JustusFT**
Ref: https://github.com/LeastAuthority/winden/issues/118

Adds the [feedback api](https://github.com/LeastAuthority/feedback-api) as well as a dummy smtp server for testing to `docker-compose.yml`. Front end integration can be found at https://github.com/LeastAuthority/winden/pull/124

~Note the feedback-api submodule is pointing to https://github.com/LeastAuthority/feedback-api/pull/6 which isn't merged~

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,6 +21,35 @@ services:
...
+    depends_on:
+      - dummy-smtp-server
+    environment:
+      - SMTP_USE_TLS=false
+      - SMTP_USERNAME=foo@foobar.in
+      - SMTP_PASSWORD=barbazquux
+  dummy-smtp-server:
+    image: python:latest
+    command:
+      ["python", "-m", "smtpd", "-n", "-c", "DebuggingServer", "0.0.0.0:2500"]
```
Comment (donpui) [inline docker-compose.yml:48]: We can switch to aiosmtpd, https://aiosmtpd.readthedocs.io/en/latest/cli.html as smtpd will depreciate.

File: feedback-api/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.19
```
Comment (donpui) [inline feedback-api/Dockerfile:1]: Can we use one common Dockerfile https://github.com/LeastAuthority/feedback-api/blob/dev/Dockerfile? Changes could be combined.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,6 @@
 [submodule "client/vendor/wormhole-william"]
 	path = client/vendor/wormhole-william
 	url = git@github.com:LeastAuthority/wormhole-william.git
+[submodule "feedback-api/feedback-api"]
+	path = feedback-api/feedback-api
```
Comment (donpui) [inline .gitmodules:5]: Isn't better to put submodule under client/vendor directory to have unified place?

Comment (JustusFT) [reply .gitmodules:5]: The idea I have with the codebase is that all top-level services reside in the project root. If something needs to be built with alongside the service (such as a library, or in this case a wasm module), then we put it in the `<service>/vendor` folder

File: dummy-smtp-server/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FROM python:latest
```
Comment (donpui) [inline dummy-smtp-server/Dockerfile:1]: Do we really need this separate Dockerfile for dummy-smtp-server. We can reuse from: https://github.com/LeastAuthority/feedback-api/tree/add-docker-composer/docker/smtp-server

**Review by donpui — APPROVED** [review]
Looks ok now. Just thinking, should we merge this to main or to add-feedback-api-form branch (and when that is finished, merge both to main).

**Comment by JustusFT** [comment]
We should merge this now, https://github.com/LeastAuthority/winden/pull/124 can be merged later once it's ready

**Pull request merged**